### PR TITLE
Solve issue #127 and add more tests

### PR DESCRIPTION
**Pull request by lbenet**

### CODE REVIEW

**Comment by lbenet** [comment]
Note that `DecoratedInterval(2//1,1//2)` throws an `ArgumentError`, since there are no `NaN`s for `Rational{T}`.

File: src/decorations/intervals.jl [hunk 1]
```diff
@@ -92,9 +98,9 @@ macro decorated(ex...)
     local x
 
     if length(ex) == 1
-        x = :(@interval($(esc(ex[1]))))
+        x = :($(esc(ex[1])))
```
Comment (dpsanders) [inline src/decorations/intervals.jl]: Won't this fail for an argument like pi? (not checked)

Comment (lbenet) [inline src/decorations/intervals.jl]: You may be right, though tests pass. I'll revert that change, because it is actually not needed. Thanks!

**Comment by lbenet** [comment]
Incidentally, tests in julia 0.5 simply get stuck in the decoration tests! I'm trying to understand what's going on there...

**Comment by lbenet** [comment]
In Julia 0.5, `DecoratedInterval(1//1, 1//10)` yields a weird error, with the first few lines being:

```julia
julia> DecoratedInterval(1//1, 1//10,com)
"WARNING: An error occured during inference. Type inference is now partially disabled."
OverflowError()
...
```

I am disabling the last two tests, which are somehow related...

**Comment by lbenet** [comment]
Tests are passing.... I don't understand codecov/patch

**Comment by dpsanders** [comment]
LGTM, thanks! I think codecov/patch refers to the fact that the coverage *of your new patch* (i.e. of the changes in this PR) has coverage which is lower than that of the project overall.

**Pull request merged**